minos/common/model/dynamic/diff.py

```diff
@@ -10,90 +10,180 @@
 )
 
 import logging
+from collections import (
+    defaultdict,
+)
 from typing import (
-    TYPE_CHECKING,
+    Any,
+    Generic,
+    Iterable,
+    Iterator,
     Optional,
+    Type,
+    TypeVar,
+    Union,
+    get_args,
+)
+from uuid import (
+    uuid4,
 )
 
+from ..abc import (
+    Model,
+)
+from ..actions import (
+    Action,
+)
+from ..fields import (
+    Field,
+)
+from ..types import (
+    ModelType,
+)
 from .bucket import (
     BucketModel,
 )
 
-if TYPE_CHECKING:
-    from ..abc import (
-        Model,
-    )
-    from ..fields import (
-        Field,
-    )
-
 logger = logging.getLogger(__name__)
 
+T = TypeVar("T")
 
-class FieldsDiff(BucketModel):
-    """Aggregate Difference class."""
+
+class Diff(Model, Generic[T]):
```

Comment:
Can we call this `FieldDiff` so it's clearer it doesn't refer to `AggregateDiff`?

Reply:
Yes! I think it's a more explanatory name, but I have one issue related with that name, and it's the similarity with `FieldsDiff` (the container class). What do you this if I rename the second one as `FieldDiffContainer` (or something similar) to avoid confusions?

Reply:
Great! I thinks that's clearer :+1: